src/plugins/auto/auto_executable_network.cpp

```diff
@@ -36,6 +36,29 @@ IE::Parameter AutoExecutableNetwork::GetConfig(const std::string& name) const {
     IE_THROW(NotFound) << name << " not found in the ExecutableNetwork config";
 }
 
+IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name, const std::string& target_device) const {
+    if (target_device == _autoSchedule->_loadContext[ACTUALDEVICE].deviceInfo.deviceName) {
+        if (_autoSchedule->_loadContext[ACTUALDEVICE].future.valid()) {
+            _autoSchedule->_loadContext[ACTUALDEVICE].future.wait();
+        }
+        try {
+            return _autoSchedule->_loadContext[ACTUALDEVICE].executableNetwork->GetMetric(name);
+        } catch (...) {
+            return _autoSchedule->_loadContext[ACTUALDEVICE].executableNetwork->GetConfig(name);
+        }
+    }
+    if (target_device == "CPU" && _autoSchedule->_loadContext[CPU].isAlready) {
+        try {
+            return _autoSchedule->_loadContext[CPU].executableNetwork->GetMetric(name);
+        } catch (...) {
+            return _autoSchedule->_loadContext[CPU].executableNetwork->GetConfig(name);
+        }
+    }
+    auto actualDevice = _autoSchedule->_loadContext[ACTUALDEVICE].deviceInfo.deviceName;
+    IE_THROW(NotFound) << target_device << "is not the device selected by " << GetLogTag()
+                       << ". Actual selected device is " << actualDevice;
```

Comment:
maybe the error message can be optimized a little bit, there are other failure situations like the target_device load failed